src/main/java/minegame159/meteorclient/systems/modules/movement/PacketFly.java

```diff
@@ -0,0 +1,311 @@
+        if(doAntiFactor.get() && checkCollisionBoxes && (mc.player.input.movementForward != 0.0 || mc.player.input.movementSideways != 0.0) && speed != 0.0) {
+            speed /= antiFactor.get();
+        }
+        double[] strafing = this.getMotion(strafeFactor.get() != false && checkCollisionBoxes ? 0.031 : 0.26);
+        for (int i = 1; i < loops.get() + 1; ++i) {
+            mc.player.setVelocity(strafing[0] * (double) i * extraFactor.get(), speed * (double) i, strafing[1] * (double) i * extraFactor.get());
+            sendPackets(mc.player.getVelocity().x, mc.player.getVelocity().y, mc.player.getVelocity().z, sendTeleport.get());
+        }
+    }
+
+    @EventHandler
+    public void onMove (PlayerMoveEvent event) {
+        if (setMove.get() && flightCounter != 0) {
+            event.movement = new Vec3d(mc.player.getVelocity().x, mc.player.getVelocity().y, mc.player.getVelocity().z);
+            if (noClip.get() && checkHitBoxes()) {
+                mc.player.noClip = true;
+            }
+        }
+    }
+
+    @EventHandler
+    public void onPacketSent(PacketEvent.Send event) {
+        PlayerMoveC2SPacket packet;
+        if (event.packet instanceof PlayerMoveC2SPacket && !packets.remove(packet = (PlayerMoveC2SPacket) event.packet)) {
+            event.setCancelled(true);
+        }
+    }
+
+    @EventHandler
+    public void onPacketReceive(PacketEvent.Receive event) {
+        if (event.packet instanceof PlayerPositionLookS2CPacket && !(mc.player == null || mc.world == null)) {
+            BlockPos pos = new BlockPos(mc.player.getPos().x, mc.player.getPos().y, mc.player.getPos().z);
+            PlayerPositionLookS2CPacket packet = (PlayerPositionLookS2CPacket) event.packet;
+            if (mc.player.isAlive() && mc.world.isChunkLoaded(mc.world.getChunk(pos).getPos().x, mc.world.getChunk(pos).getPos().z) && !(mc.currentScreen instanceof DownloadingTerrainScreen) && clearIDs.get()) {
+                teleportmap.remove(packet.getTeleportId());
+            }
+            if (setYaw.get()) {
+                ((PlayerPositionLookS2CPacketAccessor) event.packet).setPitch(mc.player.pitch);
+                ((PlayerPositionLookS2CPacketAccessor) event.packet).setYaw(mc.player.yaw);
+            }
+            if (setID.get()) {
+                teleportID = packet.getTeleportId();
+            }
+        }
+    }
+
+    private boolean checkHitBoxes() {
+        return !(mc.world.getBlockCollisions(mc.player, mc.player.getBoundingBox().expand(-0.0625,-0.0625,-0.0625)).count() == 0);
+    }
+
+    private boolean resetCounter(int counter) {
+        if (++flightCounter >= counter) {
+            flightCounter = 0;
+            return true;
+        }
+        return false;
+    }
+
+    private double[] getMotion(double speed) {
```

Comment:
Maybe this can be replaced for a method from PlayerUtils class? Not sure but it seems to do similar thing.